src/Xamarin.Android.Build.Tasks/Xamarin.Android.Common.targets

```diff
@@ -431,6 +432,13 @@ Copyright (C) 2011-2012 Xamarin. All rights reserved.
 *******************************************
 -->
 
+<PropertyGroup Condition="'$(AndroidBuildApplicationPackage)' == 'True'">
```

Comment:
When we build with `xbuild /p:AndroidBuildApplicationPackage=True`, `$(AndroidBuildApplicationPackage)` will be set for *all* projects, including library projects.

Consequently, I believe this `Condition` should also check for `$(AndroidApplication)`, so that it only operates on Application projects, not library projects:

    <PropertyGroup Condition=" '$(AndroidApplication)' == 'True' And '$(AndroidBuildApplicationPackage)' == 'True' ">